File: 240322/그 요일은/the-day-of-the-day.java
import java.util.Scanner;

public class Main {
    public static int days(int m, int d){
        int totalDays = 0;
        int[] date = new int[]{0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

        for( int i = 1 ; i < m ; i++){
            totalDays += date[i];
        }
        totalDays += d;

        return totalDays;

    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();
        String A = sc.next();
        //                            0      1      2     3       4      5      6
        String[] day = new String[]{"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};

        int elapsed = days(m2, d2) - days(m1, d1);

        int idx = 0;

        for( int i = 0 ; i < 7 ; i++){
            if(day[i].equals(A))
                idx = i;
        } // 입력받은 요일의 위치를 구함

        if( elapsed < idx )
            System.out.print(0);
        else{
            System.out.print((elapsed-idx)/7+1);
        }
    }
}

File: 240625/그 요일은/the-day-of-the-day.py
m1, d1, m2, d2 = map(int, input().split())
a = input()

date = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31] # 각 월의 날짜를 미리 기록
day = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'] # 요일, 1월 1일이 월요일이므로 Mon의 인덱스는 1
a = day.index(a) # 입력받은 요일을 day의 인덱스로 바꾸기(Mon이면 0, Sat이면 5)

def numOfDays(m, d):
    returnDate = 0
    for i in range(1, m):
        returnDate += date[i]
    returnDate += d
    return returnDate

start = numOfDays(m1, d1)
end = numOfDays(m2, d2)
cnt = 0

for i in range(start, end + 1):
    if(i % 7 == a):
        cnt += 1
"""
day_diff = numOfDays(m2, d2) - numOfDays(m1, d1)
cnt = day_diff // 7 ## 일단 이 만큼은 등장했고, 남은 일수동안 요일이 한 번 더 있는지 여부를 확인하면 됨

end = (numOfDays(m2, d2)) % 7
if end == 0:
    cnt += 1
elif 1 <= a <= end:
    cnt += 1
"""

print(cnt)